src/observable.rs

```diff
@@ -0,0 +1,75 @@
+use crate::Sealed;
+use core::marker::PhantomData;
+
+pub trait IntoObservationToken: Sized + crate::Sealed {
+    type Peripheral;
+    fn into_ot(self) -> ObservationToken<Self::Peripheral>;
+}
+
+/// A struct to hold peripherals which are to be observed.
+///
+/// This prevents the observed peripheral from being consumed. Thus
+/// preventing things like a an observed gpio pin changing mode or an opamp from
+/// being disabled. This makes sure the underlaying peripheral will not
+/// change mode into something that is not compatible with what ever may be observing it.
+pub struct Observed<P, const OBSERVER_COUNT: usize> {
+    peripheral: P,
+}
+
+impl<P, const OBSERVER_COUNT: usize> Observed<P, OBSERVER_COUNT> {
+    /// Release the observation of this peripheral
+    ///
+    /// This returns the underlaying perpheral type. Since it is no longer
+    /// observed, you are once again free to do what you want with it.
+    pub fn release(self, _data: [ObservationToken<P>; OBSERVER_COUNT]) -> P {
+        self.peripheral
+    }
+}
```

Comment:
You should definitely constrain `P` to `Observable`!

Reply:
Sorry for dumb question, but why? The only way to construct an `Observed` is with a `P` that is `Observable`, right? Just curious :)

Reply:
Not dumb at all. In all of these cases we *want* `P` to be `Observable`. It may be true that right now it just so happens that the only way to encounter these interfaces is when `P` is `Observable`, but I would consider that to be a weak invariance given that we are actively making significant changes that could silently break things. Having these additional constraints may catch other design mistakes.

Up to you, but I prefer erring on the side of caution at the expense of additional verbosity.

Reply:
Thanks! Makes sense